### PR TITLE
Correctly handle wildcards in workload entry addresses

### PR DESCRIPTION
**Pull request by ryantking**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1732

### CODE REVIEW

File: pkg/mesh-networking/translation/istio/destination/federation/federation_translator.go [hunk 1]
```diff
@@ -395,22 +396,27 @@ func getHostnameSuffix(hostname string) string {
...
+	seenAddrTypes := ksets.NewString()
 	for _, workloadEntry := range workloadEntries {
 		// only ipv6 addresses will have 2 or more colons, reference: https://datatracker.ietf.org/doc/html/rfc5952
 		if strings.Count(workloadEntry.Address, ":") >= 2 {
-			foundIpv6 = true
-		}
-		if ip := net.ParseIP(workloadEntry.Address); ip == nil {
-			foundHostname = true
+			seenAddrTypes.Insert("ipv6")
+		} else if ip := net.ParseIP(workloadEntry.Address); ip != nil {
```
Comment (Sodman) [inline pkg/mesh-networking/translation/istio/destination/federation/federation_translator.go:404]: should this also use `GetAddress()` or does that mess up the return value?

File: changelog/v1.1.0-beta37/destination_wildcards.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo-mesh/issues/1732
+    description: Check for wildcarcs in WorkloadEntry addresses and handle accordingly.
```
Comment (Sodman) [inline changelog/v1.1.0-beta37/destination_wildcards.yaml:4]: Nit: typo
```suggestion
    description: Check for wildcards in WorkloadEntry addresses and handle accordingly.
```

**Review by Sodman — COMMENTED** [review]
Can we add a test that verifies this / so that it's at least "Documented" in code as desired behavior?

File: pkg/mesh-networking/translation/istio/destination/federation/federation_translator_test.go [hunk 3]
```diff
@@ -410,4 +410,28 @@ var _ = Describe("FederationTranslator", func() {
...
+			{
+				Address: "*.mygroovyhost.net",
+			},
+		}
+		res, err := federation.ResolutionForEndpointIpVersions(workloadEntries)
+		Expect(err).To(BeNil())
+		Expect(res).To(Equal(networkingv1alpha3spec.ServiceEntry_NONE))
+	})
+
+	It("should error if there is a mix of wildcard and other adderss types", func() {
```
Comment (Sodman) [inline pkg/mesh-networking/translation/istio/destination/federation/federation_translator_test.go:425]:
```suggestion
	It("should error if there is a mix of wildcard and other address types", func() {
```

**Review by Sodman — APPROVED** [review]
LGTM

File: pkg/mesh-networking/translation/istio/destination/federation/federation_translator.go [hunk 4]
```diff
@@ -395,22 +396,29 @@ func getHostnameSuffix(hostname string) string {
 func ResolutionForEndpointIpVersions(
 	workloadEntries []*networkingv1alpha3spec.WorkloadEntry,
 ) (networkingv1alpha3spec.ServiceEntry_Resolution, error) {
-	var foundHostname bool
-	var foundIpv6 bool
+	seenAddrTypes := ksets.NewString()
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/istio/destination/federation/federation_translator.go:399]: Using a string set with hardcoded string names is prone to coding errors. I recommended using boolean variables instead, or at least assign the string keys to variables that can be reused.

**Pull request merged**